site/viamguides/src/yolo-training/yolo-training.md

```diff
@@ -0,0 +1,307 @@
+author: HipsterBrown
+id: yolo-training
+summary: Train and deploy a custom YOLO object detection model
+categories: Getting-Started, Developer
+environments: web
+status: Published 
+feedback link: https://github.com/viam-devrel/viamcodelabs/issues
+tags: Getting Started, Developer
+
+# Train & run a custom YOLO model
+<!-- ------------------------ -->
+## Overview 
+Duration: 1
+
+In the world of computer vision, [YOLO](https://docs.ultralytics.com/#yolo-a-brief-history) (You Only Look Once) has revolutionized object detection with its speed and accuracy. Unlike traditional methods that scan images multiple times at different scales, YOLO divides images into a grid and predicts bounding boxes and class probabilities in a single pass – hence the name "You Only Look Once."
+This approach makes YOLO exceptionally fast while maintaining high accuracy, perfect for real-time applications.
+Why use YOLO for your projects? It offers several key advantages:
+
+- Speed: YOLO processes images in real-time (up to 45-155 FPS depending on the model variant)
+- Accuracy: It achieves competitive performance on standard detection benchmarks
+- Versatility: Works well with objects of various sizes and in complex scenes
+- Resource efficiency: Lighter variants can run on resource-constrained devices
+- Active community: Continuous improvements and extensive documentation
+
+Custom training scripts for YOLO, like the one we'll use in this codelab, exist because while pre-trained models are powerful, they only recognize common objects they were trained on. Custom training allows you to:
+
+- Detect specific objects unique to your application
+- Optimize for your particular use case and environmental conditions
+- Fine-tune performance for your specific hardware constraints
+- Control the training process and experiment with different parameters
+
+In this codelab, we'll use the Ultralytics [YOLOv8](https://docs.ultralytics.com/models/yolov8/) implementation, which provides a good balance of accurancy and efficiency in object detection, and we'll export to ONNX format for optimal deployment on various hardware platforms.
```

Comment:
Perhaps clarify what ONNX format is here, or at least on the first mention in the remaining part of this codelab.